clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml

```diff
@@ -41,3 +40,191 @@
+      In: "CUDA::cuda_driver"
+      Out: ""
+      MatchMode: Full
+      RuleId: "remove_CUDA_driver"
+
+# libsycl.so is linked by default as alternative to cuda_driver library, so removing here
+- Rule: rule_target_link_libraries_cuda_rt
+  Kind: CMakeRule
+  Priority: Fallback
+  CmakeSyntax: cudart_target_link_libraries
+  In: target_link_libraries(${CUDALibraries})
+  Out: target_link_libraries(${CUDALibraries})
+  Subrules:
+    CUDALibraries:
+      In: "CUDA::cudart"
+      Out: ""
+      MatchMode: Full
+      RuleId: "remove_CUDA_cudart"
+
+- Rule: rule_target_link_libraries_cudnn
+  Kind: CMakeRule
+  Priority: Fallback
+  CmakeSyntax: cudnn_target_link_libraries
+  In: target_link_libraries(${CUDALibraries})
+  Out: target_link_libraries(${CUDALibraries})
+  Subrules:
+    CUDALibraries:
+      In: "CUDNN::cudnn"
+      Out: libdnn.so
+      MatchMode: Full
+      RuleId: "replace_cudnn_with_libdnn"
+
+# SYCL does not have mapping for nvrtc currently, so removing here
+- Rule: rule_target_link_libraries_nvrtc
+  Kind: CMakeRule
+  Priority: Fallback
+  CmakeSyntax: nvrtc_target_link_libraries
+  In: target_link_libraries(${CUDALibraries})
+  Out: target_link_libraries(${CUDALibraries})
+  Subrules:
+    CUDALibraries:
+      In: "CUDA::nvrtc"
+      Out: ""
+      MatchMode: Full
+      RuleId: "remove_nvrtc"
+  
+# SYCL does not have mapping for nvToolsExt currently, so removing here
+- Rule: rule_target_link_libraries_nvToolsExt
+  Kind: CMakeRule
+  Priority: Fallback
+  CmakeSyntax: nvToolsExt_target_link_libraries
+  In: target_link_libraries(${CUDALibraries})
+  Out: target_link_libraries(${CUDALibraries})
+  Subrules:
+    CUDALibraries:
+      In: "CUDA::nvToolsExt"
+      Out: ""
+      MatchMode: Full
+      RuleId: "remove_nvToolsExt"
+
+# SYCL does not have mapping for libnvinfer currently, so removing here
+- Rule: rule_target_link_libraries_libnvinfer
+  Kind: CMakeRule
+  Priority: Fallback
+  CmakeSyntax: libnvinfer_target_link_libraries
+  In: target_link_libraries(${CUDALibraries})
+  Out: target_link_libraries(${CUDALibraries})
+  Subrules:
+    CUDALibraries:
+      In: libnvinfer.so
+      Out: ""
+      MatchMode: Full
+      RuleId: "remove_libnvinfer"
+
+# SYCL does not have mapping for libnvonnxparser currently, so removing here
+- Rule: rule_target_link_libraries_libnvonnxparser
+  Kind: CMakeRule
+  Priority: Fallback
+  CmakeSyntax: libnvonnxparser_target_link_libraries
+  In: target_link_libraries(${CUDALibraries})
+  Out: target_link_libraries(${CUDALibraries})
+  Subrules:
+    CUDALibraries:
+      In: libnvonnxparser.so
+      Out: ""
+      MatchMode: Full
+      RuleId: "remove_libnvonnxparser"
+
+- Rule: rule_target_link_libraries_libstdc++
+  Kind: CMakeRule
+  Priority: Fallback
+  CmakeSyntax: libstdc++_target_link_libraries
+  In: target_link_libraries(${StaticLibraries})
+  Out: target_link_libraries(${StaticLibraries})
+  Subrules:
+    StaticLibraries:
+      In: -static-libstdc++
```

Comment:
Pls add comments to explain why "-static-libstdc++" should be removed.

Reply:
Added in helper file in seperate PR: https://github.com/oneapi-src/SYCLomatic/pull/1544